Docs 7754 dwformats duke (#1319)
mulesoft/docs-mule-runtime-3: modules/ROOT/pages/dataweave-formats-xml.adoc

The JSON input serves as the payload to the DataWeave source. Notice that the
`name` keys in the array are repeated.

[source,json,linenums]
----
{
  "friends": [
    {"name": "Mariano"},
    {"name": "Shoki"},
    {"name": "Tomo"},
    {"name": "Ana"}
  ]
}
----

==== Source

The DataWeave script selects the value of the `friends` key.

[source,dataweave,linenums]
----
%dw 2.0
output application/xml
---
friends: {
    (payload.friends)
}
----

==== Output

The output represents the `name` keys as XML elements.

[source,xml,linenums]
----
<?xml version='1.0' encoding='UTF-8'?>
<friends>
  <name>Mariano</name>
  <name>Shoki</name>
  <name>Tomo</name>
  <name>Ana</name>
</friends>
----

See also, xref:dataweave-cookbook-output-self-closing-xml-tags.adoc[Example: Outputting Self-closing XML Tags].

// CONFIG PROPS ///////////////////////////////////////////////////////

[[properties]]
== Configuration Properties

DataWeave supports the following configuration properties for the XML format.

=== Reader Properties

The XML reader accepts properties that provide instructions for reading input data.

[cols="1,1,1,3a", options="header"]
|===
| Parameter | Type | Default | Description

> |`collectionPath`  |`String`| None | Path to the document where the collection is located. It specifies a path expression that identifies the location of the elements to stream. See <<example1>>.

Author: ```suggestion
|`collectionPath`  |`String`| `null` | Path to the document where the collection is located. It specifies a path expression that identifies the location of the elements to stream. See <<example1>>.
```

accepted in an XML attribute. Available since Mule 4.2.1.
| `maxEntityCount` | `Number` | `1` | The maximum number of entity expansions.
The limit is in place to avoid Billion Laughs attacks.
| `nullValueOn` | `String` | `blank` | Indicates whether to read an element with empty or blank text
as a `null` value. Valid values are `empty`, `none`, or `blank`. See <<example3>> and <<example4>>.
|`optimizeFor` |`String`| `'speed'` | Indicates how to configure the XML parser. Valid values are `speed` and `memory`.
|`streaming` |`Boolean` | `false` | Indicates whether to stream input (use only if entries are accessed sequentially). Valid Options are `true` or `false`. See <<example1>>.
|`supportDtd` |`Boolean` | `true` |  Enable or disable DTD handling. Disabling skips (and does not process) internal and external subsets. Valid Options are `true` or `false`.
|===

=== Writer Properties

The XML writer accepts properties that provide instructions for writing output data.

[cols="1,1,1,3a", options="header"]
|===
| Parameter | Type | Default | Description
| `bufferSize` | `Number` | `8192` | Size of the buffer writer.

> | `encoding` | `String` | None |Encoding for the writer to use.

Author: ```suggestion
| `encoding` | `String` | `null` |Encoding for the writer to use.
```

| `deferred` | `Boolean` | `false` |Property for deferred output.
  Valid values are `true` or `false`.
| `escapeCR` | `Boolean` | `false` | Whether to escape a carriage return (CR).
Valid values are `true` or `false`. Available since Mule 4.2.1.

> | `indent` | `Boolean` | `true` |Indicates whether to indent the output.

Author: ```suggestion
| `indent` | `Boolean` | `true` |Indicates whether to indent the code for better readability or to compress it into a single line. 
```

Valid values are `true` or `false`.

> | `inlineCloseOn` | `String` | `empty` |When the writer should use inline close

Author: ```suggestion
| `inlineCloseOn` | `String` | `empty` |Indicates when the writer uses inline close tag or use open close tags.
```

> tag. Valid values are `empty` or `none`.

Author: ```suggestion
Valid values are `empty` or `none`.
```

> | `onInvalidChar` | `String` | None |Valid values are `base64`, `ignore`, or `none`.

Author: ```suggestion
| `onInvalidChar` | `String` | `null` | Valid values are `base64`, `ignore`, or `none`.
```
